src/Backend/Modules/MediaLibrary/Domain/MediaItem/StorageType.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Backend\Modules\MediaLibrary\Domain\MediaItem;
+
+final class StorageType
+{
+    // Possible MediaItem types
+    const EXTERNAL = 'external';
+    const LOCAL = 'local';
+    const YOUTUBE = 'youtube';
+    const VIMEO = 'vimeo';
+    const POSSIBLE_VALUES = [
+        self::EXTERNAL,
+        self::LOCAL,
+        self::YOUTUBE,
+        self::VIMEO,
+    ];
+    const POSSIBLE_VALUES_FOR_MOVIE = [
+        self::YOUTUBE,
+        self::VIMEO,
+    ];
+
+    /** @var string */
+    private $mediaItemStorageType;
+
+    /**
+     * @param string $mediaItemStorageType
+     */
+    private function __construct(string $mediaItemStorageType)
+    {
+        if (!in_array($mediaItemStorageType, self::POSSIBLE_VALUES, true)) {
+            throw new \InvalidArgumentException('Invalid value');
+        }
+
+        $this->mediaItemStorageType = $mediaItemStorageType;
+    }
+
+    /**
+     * @param string $mediaItemStorageType
+     * @return StorageType
+     */
+    public static function fromString(string $mediaItemStorageType): StorageType
+    {
+        return new self($mediaItemStorageType);
+    }
+
+    /**
+     * @return string
+     */
+    public function __toString(): string
+    {
+        return (string) $this->mediaItemStorageType;
+    }
+
+    /**
+     * @param StorageType $mediaItemStorageType
+     *
+     * @return bool
+     */
+    public function equals(StorageType $mediaItemStorageType): bool
+    {
+        if (!($mediaItemStorageType instanceof $this)) {
+            return false;
+        }
+
+        return $mediaItemStorageType == $this;
```

Comment:
`return $mediaItemStorageType->mediaItemStorageType === $this->mediaItemStorageType;`